### ISSUE TITLE
@handles in URLs trigger unwantedn notifications

### ISSUE THREAD

**Issue by tibor** [mastodon/mastodon#23275]
### Steps to reproduce the problem

1. Post a url with a handle in it (https://www.tiktok.com/@csuauftiktok)
2. The user @csuauftiktok on mastodon would get a notification although he is never tagged in the post itself

Example, I got a notification here for my account @tibor@mastodon.social: https://mastodon.social/@tiborde/109762298063815003 because @tibor is in the URL

### Expected behaviour

No notification

### Actual behaviour

User gets notification

### Specifications

Mastodon v4.0.2

**Comment by tibor** [comment]
![Image 2](https://user-images.githubusercontent.com/3136416/215153555-38adb6d1-5f71-4cfa-8fd9-4216ff1c5fd7.jpg)

**Comment by ClearlyClaire** [comment]
Good catch. This occurs in certain cases but not others, e.g. `https://foo/@bar` does not mention `@bar`, but `https://foo/?bar=@bar` does

**Comment by ClearlyClaire** [comment]
Closed by #23275
